pkg/roachprod/vm/gce/gcloud.go

```diff
@@ -1910,6 +1910,37 @@ func (p *Provider) CreateLoadBalancer(_ *logger.Logger, vms vm.List, port int) e
 	return g.Wait()
 }
 
+// ListLoadBalancers returns the list of load balancers associated with the
+// given VMs. The VMs have to be part of a managed instance group. The load
+// balancers are returned as a list of service addresses.
+func (p *Provider) ListLoadBalancers(_ *logger.Logger, vms vm.List) ([]vm.ServiceAddress, error) {
+	// Only managed instance groups support load balancers.
+	if !isManaged(vms) {
+		return nil, nil
+	}
+	project := vms[0].Project
+	clusterName, err := vms[0].ClusterName()
+	if err != nil {
+		return nil, err
+	}
+	rules, err := listForwardingRules(project)
+	if err != nil {
+		return nil, err
+	}
+
+	addresses := make([]vm.ServiceAddress, 0)
+	for _, rule := range rules {
+		cluster, resourceType, port, ok := loadBalancerNameParts(rule.Name)
+		if !ok {
+			continue
+		}
+		if cluster == clusterName && resourceType == "forwarding-rule" {
```

Comment:
True, but since other people are also creating forwarding rules outside of `roachprod` it's a bit of an extra sanity check to make sure we're not dealing with a `non-roachprod` forwarding rule. Which is also checked by the `roachprod` tag in the name.